core/string/ustring.cpp

```diff
@@ -1946,7 +1946,7 @@ Error String::append_utf8(const char *p_utf8, int p_len, bool p_skip_cr) {
 	}
 
 	(*dst++) = 0;
-	resize_uninitialized(prev_length + dst - ptr());
+	resize_uninitialized(dst - ptr());
```

Comment:
This ends up resizing the string with additional data, that is uninitialised and may cause crash and result in corrupted string data.